### DISCUSSION TITLE
TO evaluate the trained detection model, dataset format same as used for training

### DISCUSSION THREAD

**Discussion by madhavi1102** [mindee/doctr#1372] [Q&A] [answered]
I have done training of detection model on docTR using custom dataset having format images and labels.json file. The same data format do we need (images with labels.json) to be created when doing evaluating. 
 
It will be helpful if details someone provide on it.

Also, I want to understand the purpose of using pretrained=True in case of fine-tuning the detection model.
First time we can keep pretrained as True, but in second I want to retrain again with my trained checkpoint rather than the  pretrained. Does it mean, for further iterations I don't need to set  pretrained=True and use option --resume in training.

Helpful if someone clear my doubts.

Thanks

**Comment by felixT2K** [comment, marked as answer]
Hi @madhavi1102 👋🏼,

1. If you want to evaluate on your own data you can modify the https://github.com/mindee/doctr/blob/main/references/detection/evaluate_pytorch.py  script to work with your own data (replace the dataset with `DetectionDataset`) or make use of the training script `--test-only` flag https://github.com/mindee/doctr/blob/4cadd8d681130327da80a1a5cc4b4861ff481f14/references/detection/train_pytorch.py#L425C93-L425C103
2. Correct if you want to resume from your checkpoint there is no need to pass --pretrained

**Reply by madhavi1102** [reply]
Hi felixT2X,

I really appreciate it, it is indeed very helpful to me clearing my all doubts, even though I am asking the small queries.

Thank you :)
